work_queue/src/work_queue.c

```diff
@@ -3719,6 +3722,21 @@ static int check_hand_against_task(struct work_queue *q, struct work_queue_worke
 	if(w->resources->gpus.inuse + limits->gpus > overcommitted_resource_total(q, w->resources->gpus.total, 0)) {
 		ok = 0;
 	}
+		
+	//if the wall time for the worker is specified and there's not enough time for the task, then not ok
+	if(w->end_time){
+		if(t->resources_requested->wall_time > 0){
+			if(w->end_time - time(0) > t->resources_requested->wall_time) {
```

Comment:
we can simplify with:
if(w->end_time < time(0) + t->resources_requested->wall_time) {
ok = 0;
}